server/v2/store/server.go

```
@@ -63,15 +64,29 @@ func (s *Server[T]) CLICommands() serverv2.CLIConfig {
 			s.ListSnapshotsCmd(),
 			s.DumpArchiveCmd(),
 			s.LoadArchiveCmd(),
-			s.RestoreSnapshotCmd(s.appCreator),
+			s.RestoreSnapshotCmd(s.backend),
 		},
 	}
 }
 
 func (s *Server[T]) Config() any {
 	if s.config == nil || s.config.AppDBBackend == "" {
-		return DefaultConfig()
+		return root.DefaultConfig()
 	}
 
 	return s.config
 }
+
+func UnmarshalConfig(cfg map[string]any) *root.Config {
+	config := &root.Config{
+		Options: root.DefaultStoreOptions(),
+	}
+	if err := serverv2.UnmarshalSubConfig(cfg, ServerName, config); err != nil {
+		panic(fmt.Sprintf("failed to unmarshal config: %v", err))
```

let's return an error here and return down in init as well